confgenerator/logging_processors.go

```diff
@@ -364,3 +368,66 @@ end`, lua))...)
 func init() {
 	LoggingProcessorTypes.RegisterType(func() LoggingProcessor { return &LoggingProcessorExcludeLogs{} })
 }
+
+// A LoggingProcessorGceMetadataAttributes annotates logs with labels
+// corresponding to instance attributes from the GCE metadata server.
+type LoggingProcessorGceMetadataAttributes struct {
+	ConfigComponent `yaml:",inline"`
+	Include         []string `yaml:"include" validate:"dive,gt=0,glob"`
+	Exclude         []string `yaml:"exclude" validate:"dive,gt=0,glob"`
+	Static          *bool `yaml:"static" validate:"required,eq=true"`
+}
+
+func (p LoggingProcessorGceMetadataAttributes) Type() string {
+	return "internal:gce_metadata_attributes"
+}
+
+func matchesAnyGlobs(globs []string, k string) bool {
+	for _, g := range globs {
+		// Errors would have been caught by the validators.
+		if ok, _ := filepath.Match(g, k); ok {
+			return true
+		}
+	}
+	return false
+}
+
+const (
+	attributeLabelPrefix string = "compute.googleapis.com/attributes/"
+)
+
+func (p LoggingProcessorGceMetadataAttributes) Components(ctx context.Context, tag, uid string) []fluentbit.Component {
+	processorName := fmt.Sprintf("%s.%s.gce_metadata", tag, uid)
+	resource := MetadataResource
```

Comment:
This global variable exists only for testing the Prometheus receiver; you shouldn't reuse it here for a different purpose. You should just call `resourcedetector.GetResource` (and arguably that should take a context argument and check for a resource override itself)

Reply:
Now that `ResourceOverride` has been added, do we even need `MetadataResource` anymore?

Reply:
Updated to remove `MetadataResource`, and use `platform.GetResource()` instead. I couldn't put that code in `resourcedetector`, as that would result in an import cycle.